### PR TITLE
Core/Object: Range check vol. 2

### PR DESCRIPTION
**Pull request by d0d0**
**Changes proposed:**

-  implement reviews from https://github.com/TrinityCore/TrinityCore/pull/23179

**Target branch(es):** 3.3.5/master

- [x] 3.3.5

**Issues addressed:** Closes #

https://github.com/TrinityCore/TrinityCore/issues/15453 
https://github.com/TrinityCore/TrinityCore/issues/23062#issuecomment-486745022
https://github.com/TrinityCore/TrinityCore/issues/22530
https://github.com/TrinityCore/TrinityCore/issues/23205

**Tests performed:** 

Ingame with all cases from original issue 
https://github.com/TrinityCore/TrinityCore/issues/23062
https://github.com/TrinityCore/TrinityCore/issues/15453
(quests like https://github.com/TrinityCore/TrinityCore/issues/17872  https://github.com/TrinityCore/TrinityCore/issues/22530 Polishing the Helm)

### CODE REVIEW

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -2554,14 +2554,34 @@ float GameObject::GetInteractionDistance() const
...
+        case GAMEOBJECT_TYPE_MINI_GAME:
+            return 5.5555553f;
+        case GAMEOBJECT_TYPE_BINDER:
             return 10.0f;
+        case GAMEOBJECT_TYPE_CHAIR:
+        case GAMEOBJECT_TYPE_BARBER_CHAIR:
+            return 3.0f;
         case GAMEOBJECT_TYPE_FISHINGHOLE:
         case GAMEOBJECT_TYPE_FISHINGNODE:
             return 20.0f + CONTACT_DISTANCE; // max spell range
```
Comment (xvwyh) [inline src/server/game/Entities/GameObject/GameObject.cpp]: I took all the values from the client executable, so you might as well have left 100 yards there.
image
image
`GAMEOBJECT_TYPE_FISHINGNODE` is initialized to 100 yards, but `GAMEOBJECT_TYPE_FISHINGHOLE` reuses the same constructor as `GAMEOBJECT_TYPE_SPELL_FOCUS` (which is also reused by `GAMEOBJECT_TYPE_DUEL_ARBITER` and `GAMEOBJECT_TYPE_AURA_GENERATOR`, hey, compiler can optimize things however it wants) which just calls the base constructor and initializes interaction distance to default 5 yards. Which is because player never really interacts with a fishing hole, only with a fishing node, which on the server checks if there are any holes nearby and sends their loot instead of the fishing loot for the current zone.

But I won't insist, since realistically the only time this will be relevant is if a player decides to fish from a moving transport, like a ship or a turtle taxi in Northrend. 😄

**Comment by Killyana** [comment]
I tested all know cases, and everything looks fine.

**Comment by Faq** [comment]
so only range correction for GAMEOBJECT_TYPE_FISHINGNODE missing

**Comment by jackpoz** [comment]
is this PR ready to be merged ?

**Comment by xvwyh** [comment]
Thank you, github behaved this time and didn't unsub me randomly.
I don't have any other problems with this PR apart from codestyle, but I'll leave that critique to someone better versed in TC's codestyle than me. Functionality-wise, it can have my worthless imaginary seal of approval.

**Comment by Shauren** [comment]
Codestyle looks fine to me

**Comment by xvwyh** [comment]
I meant the things like comma at the end of enum values (which seems to be diligently expunged in TC, which is something I personally strongly disagree on); `case GAMEOBJECT_TYPE_FISHINGHOLE` not being dropped down to "non blizzlike values"; redefinition of the meaning of `IsWithinDistInMap` without actually having said "dist" as a parameter thus making the function name an odd choice at first glance (and IMO it's a better idea to put IsInMap, IsInPhase checks into IsAtInteractDistance, which is actually something I should do in my server too now that I think about it, brb); also I wish I had came up with a better name for IsAtInteractDistance, but WorldObject having all these similar methods like IsInRange, IsInRange2d, IsInRange3d, IsWithinDistInMap and so on already makes it confusing enough; not to mention that GameObject still has the now obsolete method IsInRange (which overloads WorldObject::IsInRange with a completely different implementation), which probably should've been replaced with IsAtInteractDistance's contents instead of adding it alongside one another, yadda yadda yadda.

But none of this detracts from the PR's intent and all this can be fixed later provided someone will care enough.

**Comment by jackpoz** [comment]
tl;dr I merged it :)
Thanks for the PR :)
